ddl/ddl.go

```diff
@@ -477,116 +450,35 @@ func (d *ddl) AlterTable(ctx context.Context, ident table.Ident, specs []*AlterS
 			return errors.Trace(err)
 		}
 	}
+
 	return nil
 }
 
-// Add a column into table
-func (d *ddl) addColumn(ctx context.Context, schema *model.DBInfo, tbl table.Table, spec *AlterSpecification, schemaMetaVersion int64) error {
-	// Find position
-	cols := tbl.Cols()
-	position := len(cols)
-	name := spec.Column.Name
-	// Check column name duplicate.
-	dc := column.FindCol(cols, name)
-	if dc != nil {
-		return errors.Errorf("Try to add a column with the same name of an already exists column.")
-	}
-	if spec.Position.Type == ColumnPositionFirst {
-		position = 0
-	} else if spec.Position.Type == ColumnPositionAfter {
-		// Find the mentioned column.
-		c := column.FindCol(cols, spec.Position.RelativeColumn)
-		if c == nil {
-			return errors.Errorf("No such column: %v", name)
-		}
-		// Insert position is after the mentioned column.
-		position = c.Offset + 1
+// AddColumn will add a new column to the table.
+func (d *ddl) AddColumn(ctx context.Context, ti table.Ident, spec *AlterSpecification) error {
```

Comment:
do we check index constraint when adding a column with unique or other extra constraints?